numpy/core/tests/test_arrayprint.py

```diff
@@ -354,6 +354,11 @@ def test_sign_spacing(self):
 
         assert_raises(TypeError, np.set_printoptions, wrongarg=True)
 
+    def test_float_overflow_nowarn(self):
+        # make sure internal computations in FloatingFormat don't
+        # warn about overflow
+        repr(array([1.0, 1e-310], dtype='f8'))
```

Comment:
Nice!

Reply:
Hmm but actually tests just finished running on my system and I don't think this triggers the problem because the `min_val < 0.0001` short-circtuits out the division clause anytime overflow could occur.

I think overflow may only be posible for float16 types under certain circumstances. `np.array([1e4, 0.1], dtype='f2')` works.